fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindDynamicSplit.java

```diff
@@ -0,0 +1,196 @@
+
+                            Plan result = ctx.root.child();
+                            LogicalDynamicSplit logicalDynamicSplit = ctx.root;
+                            LogicalRelation logicalRelation = (LogicalRelation) ((LogicalProject<?>)
+                                    ((LogicalFilter<?>) logicalDynamicSplit.child()).child()).child();
+                            return rewritePlan(logicalRelation, result, logicalDynamicSplit);
+                        })),
+                //LogicalDynamicSplit -> LogicalSubQueryAlias -> LogicalView
+                RuleType.BIND_DYNAMIC_SPLIT.build(logicalDynamicSplit(logicalSubQueryAlias(logicalView()))
+                        .thenApply(ctx -> {
+
+                            Plan result = ctx.root.child();
+                            LogicalDynamicSplit<LogicalSubQueryAlias<LogicalView<Plan>>> logicalDynamicSplit = ctx.root;
+                            LogicalView logicalView = (LogicalView) ((LogicalSubQueryAlias<?>) logicalDynamicSplit
+                                    .child()).child();
+
+                            return rewritePlan(logicalView, result, logicalDynamicSplit);
+                        })),
+                // LogicalDynamicSplit -> LogicalCatalogRelation
+                RuleType.BIND_DYNAMIC_SPLIT.build(logicalDynamicSplit(logicalRelation()).thenApply(ctx -> {
+                    Plan result = ctx.root.child();
+                    LogicalDynamicSplit<LogicalRelation> logicalDynamicSplit = ctx.root;
+                    LogicalRelation logicalCatalogRelation = ctx.root.child();
+                    if (null == logicalCatalogRelation) {
+                        return result;
+                    }
+                    return rewritePlan(logicalCatalogRelation, result, logicalDynamicSplit);
+                })));
+    }
+
+    private LogicalCatalogRelation convertAndCheckLogicalRelation(LogicalRelation logicalRelation) {
+        if (logicalRelation instanceof LogicalCatalogRelation) {
+            return (LogicalCatalogRelation) logicalRelation;
+        }
+        return null;
+    }
+
+    private Plan rewritePlan(LogicalView logicalView, Plan result,
+                             LogicalDynamicSplit logicalDynamicSplit) {
+        Range range = logicalDynamicSplit.getRange();
+        SplitColumnInfo splitColumnInfo = logicalDynamicSplit.getSplitColumnInfo();
+        String splitColumnName = splitColumnInfo.getColumnName();
+        TableName splitTable = splitColumnInfo.getTableNameInfo().transferToTableName();
+        View view = logicalView.getView();
+        logicalDynamicSplit.setReplaced(Boolean.TRUE);
+        if (!view.getName().equalsIgnoreCase(splitTable.getTbl())) {
+            return result;
+        }
+        if (!view.getDatabase().getFullName().equals(splitTable.getDb())) {
+            return result;
+        }
+        if (!view.getDatabase().getCatalog().getName().equals(splitTable.getCtl())) {
+            return result;
+        }
```

Comment:
I think logicalDynamicSplit is not replaced in this place, because the view name maybe different to the logicalDynamicSplit and return the child

Reply:
If the view names are inconsistent, it indicates that this part should not be replaced. When performing an insert into select operation involving multiple tables, only the tables that meet the criteria need to have the corresponding filter condition added.